<!DOCTYPE html>
<html>
  <head>
    <title>Introduction to Genome Annotation</title>
        <meta http-equiv="Content-Type" content="text/html;
						 charset=UTF-8"/>
	<style type="text/css">
	        @import
	        url(https://fonts.googleapis.com/css?family=Yanone+Kaffeesatz);
		      @import
	        url(https://fonts.googleapis.com/css?family=Droid+Serif:400,700,400italic);
		      @import
	        url(https://fonts.googleapis.com/css?family=Consolas:400,700,400italic);

		body { font-family: 'Droid Serif'; }
		h1, h2, h3 {
		font-family: 'Yanone Kaffeesatz';
		font-weight: normal;
		}
		.remark-code, .remark-inline-code { font-family:
		'Consolas'; }
	</style>
  </head>
  <body>
    <textarea id="source">
      class: center, middle

      # Introduction to Genome Annotation

      Jason Stajich
      UC Riverside

      http://lab.stajich.org
      
      http://stajichlab.github.io/GenomeAnnotation
      
      ---
      # Genome As a Book

      ```
      CHAPTER V

      That same evening there was an animated discussion among the squadron's
      officers in Denisov's quarters.

      "And I tell you, Rostov, that you must apologize to the colonel!" said a
      tall, grizzly-haired staff captain, with enormous mustaches and many
      wrinkles on his large features, to Rostov who was crimson with
      excitement.

      The staff captain, Kirsten, had twice been reduced to the ranks for
      affairs of honor and had twice regained his commission.

      "I will allow no one to call me a liar!" cried Rostov. "He told me I
      lied, and I told him he lied. And there it rests. He may keep me on duty
      every day, or may place me under arrest, but no one can make me
      apologize, because if he, as commander of this regiment, thinks it
      beneath his dignity to give me satisfaction, then..."

      "You just wait a moment, my dear fellow, and listen," interrupted the
      staff captain in his deep bass, calmly stroking his long mustache.
      ```
 
      ---
      # No word boundaries it begins to be hard to read

      ```
      CHAPTERVThatsameeveningtherewasananimateddiscussionamongthesquadron'soffice
      rsinDenisov'squarters."AndItellyou,Rostov,thatyoumustapologizetothecolonel!
      "saidatall,grizzly-hairedstaffcaptain,withenormousmustachesandmanywrinkleso
      nhislargefeatures,toRostovwhowascrimsonwithexcitement.Thestaffcaptain,Kirst
      en,hadtwicebeenreducedtotheranksforaffairsofhonorandhadtwiceregainedhiscomm
      ission."Iwillallownoonetocallmealiar!"criedRostov."HetoldmeIlied,andItoldhi
      mhelied.Andthereitrests.Hemaykeepmeondutyeveryday,ormayplacemeunderarrest,b
      utnoonecanmakemeapologize,becauseifhe,ascommanderofthisregiment,thinksitben
      eathhisdignitytogivemesatisfaction,then...""Youjustwaitamoment,mydearfellow
      ,andlisten,"interruptedthestaffcaptaininhisdeepbass,calmlystrokinghislongmu
      stache.
      ```
 
      ---
      # With no punctionation harder to read

      ```
      CHAPTERVThatsameeveningtherewasananimateddiscussionamongthesquadronsofficer
      sinDenisovsquartersAndItellyouRostovthatyoumustapologizetothecolonelsaidata
      llgrizzlyhairedstaffcaptainwithenormousmustachesandmanywrinklesonhislargefe
      aturestoRostovwhowascrimsonwithexcitementThestaffcaptainKirstenhadtwicebeen
      reducedtotheranksforaffairsofhonorandhadtwiceregainedhiscommissionIwillallo
      wnoonetocallmealiarcriedRostovHetoldmeIliedandItoldhimheliedAndthereitrests
      Hemaykeepmeondutyeverydayormayplacemeunderarrestbutnoonecanmakemeapologizeb
      ecauseifheascommanderofthisregimentthinksitbeneathhisdignitytogivemesatisfa
      ctionthenYoujustwaitamomentmydearfellowandlisteninterruptedthestaffcaptaini
      nhisdeepbasscalmlystrokinghislongmustache
      ```

      ---
      # No capitalization even harder to parse

      ```
      chaptervthatsameeveningtherewasananimateddiscussionamongthesquadronsofficers
      indenisovsquartersanditellyourostovthatyoumustapologizetothecolonelsaidatall
      grizzlyhairedstaffcaptainwithenormousmustachesandmanywrinklesonhislargefeatu
      restorostovwhowascrimsonwithexcitementthestaffcaptainkirstenhadtwicebeenredu
      cedtotheranksforaffairsofhonorandhadtwiceregainedhiscommissioniwillallownoon
      etocallmealiarcriedrostovhetoldmeiliedanditoldhimheliedandthereitrestshemayk
      eepmeondutyeverydayormayplacemeunderarrestbutnoonecanmakemeapologizebecausei
      fheascommanderofthisregimentthinksitbeneathhisdignitytogivemesatisfactionthe
      nyoujustwaitamomentmydearfellowandlisteninterruptedthestaffcaptaininhisdeepb
      asscalmlystrokinghislongmustache
      ```
            
      
      ---
      # Finding the words on the page

      
      __Helpful to have markings to know what are words, or where
      words start or end.__

      
      ---
      # Comparing texts to known words

      Search a database of known words against our text might pick out
      the word boundaries. Starting with most frequent words.

      * and     22147   3.91%
      * to      16713   2.95%
      * of      14981   2.65%
      * a       10509   1.86%
      * he      9746    1.72%
      * in      8786    1.55%
      * his     7963    1.41%
      * that    7794    1.38%
      * was     7328    1.29%
      * with    5691    1.00%
      * had     5353    0.95%
      * it      5143    0.91%
      * her     4695    0.83%
      * not     4655    0.82%
      * him     4563    0.81%
      
      ---
      # Frequency of words above certain length

      * which   1996    1.24%   5
      * prince  1879    1.17%   6
      * pierre  1775    1.11%   6
      * there   1453    0.90%   5
      * their   1435    0.89%   5
      * would   1362    0.85%   5
      * could   1111    0.69%   5
      * natasha 1089    0.68%   7
      * andrew  1037    0.65%   6
      * about   1016    0.63%   5
      * himself 1005    0.63%   7
      * princess        914     0.57%   8

      ---
      # Frequency of certain words

      (order, word, number of times it appears, frequency)

      * 2094	vodka	24	0.00%
      * 1283	mustache	44	0.01%
      * 3917	mustaches	11	0.00%
      * 12159	mustached	1	0.00%
      * 12320	blackmustached	1	0.00%
      * 578	captain	113	0.02%
      * 9497	captaincy	2	0.00%
      * 9553	captains	2	0.00%


      ---
      # Using frequently occuring words

      ***Match frequently occurring words back against the text to
      pull out there incidence and annotate the region.***

      ```
      chaptervthatsameeveningtherewasananimateddiscussionamongthesquadronsofficers
      indenisovsquartersanditellyourostovthatyoumustapologizetothecolonelsaidatall
      grizzlyhairedstaffcaptainwithenormous [mustaches] andmanywrinklesonhislargefeatu
      restorostovwhowascrimsonwithexcitementthestaffcaptainkirstenhadtwicebeenredu
      cedtotheranksforaffairsofhonorandhadtwiceregainedhiscommissioniwillallownoon
      etocallmealiarcriedrostovhetoldmeiliedanditoldhimheliedandthereitrestshemayk
      eepmeondutyeverydayormayplacemeunderarrestbutnoonecanmakemeapologizebecausei
      fheascommanderofthisregimentthinksitbeneathhisdignitytogivemesatisfactionthe
      nyoujustwaitamomentmydearfellowandlisteninterruptedthestaffcaptaininhisdeepb
      asscalmlystrokinghislong [mustache]

      ```
      
      ---
      # From frequencies to models

      1. Development of a decoder for DNA into genes requires building
      models of what is recognized by the transcription and
      translation machinery
      
      2. Data mining approach to develop statistical probabilities
      about gene, exon length, sequence bias at single, doublet, or
      triplet nucleotides

      3. 
      
      ---
      # Central Dogma

      ![CentralDogma](img/centraldogma.gif)

      ---
      # Eukaryotic gene structure

      ![EukGeneStructure](img/Gene_structure_eukaryote_2_annotated.svg.png)

      ---
      # Genome size
      ![GenomeSizeGeneCount](img/Yandell_Fig1.png)

      Yandell and Ence 2012, Nat Rev Genetics

      ---
      # Patterns and rules in the data
      
      1. Codons are triplets of DNA letters.
       - Coding regions should look different than non-coding regions
      in frequency of 2-letter and 3-letter words
      2. Coding region starts with ATG (AUG) and end with stop codon
       (TAA TAG TGA)
      3. Splice sites for Introns
     

      ---
      # Ab initio gene finders

      - Built on training sets of known genes
      - Estimate frequencies of which of the 61 codons are used, and the transitions
      between codons
      - Incorporate splice site information to predict when exons end
      and introns start
      - Exon and intron length is also a parameter estiated from the
      data
      - Together these observed statistics can be parametrized into an
      Hidden Markov Model
      
      ---
      #Gene finding HMM

      ![GlimmerHMM](img/GlimmerHMM_Fig3.gif)

      ---
      # Ab initio Gene finding HMM

      1. Each box is a particular state (e.g. Exon, Intron,
      Intergenic)
      2. From training data, estimate frequency which will jump to
      gene start based on whether there is an ATG.
      2. From training data, can estimate the frequency by which jump
      to other states (e.g. from exon to intron) which are based on
      the current base/codon frequency, the length of feature, and
      window of sequence (are we near a likely splice site).
      3. Overall consider all paths through the data (e.g. transitions
      to consecutive states) but only keep the best scoring ones. 

      ---
      # Evidence based gene predictions

      1. Generate alignments of experimental sequence data to the
      genomic locus
      2. Transcript support: Expressed Sequence Tags (ESTs) or more commonly now
      RNASeq data or full length transcripts
      3. Homology based prediction: Protein sequences either inferred from related species
      proteins, inferred from a translation of RNASeq/ESTs 

      ---
      # Generating evidence predictions
      1. Need to align evidence to the genome
      2. Use BLASTX initially to align protein to genome
      3. But isn't splice-aware, need other tools which do this
      4. Exonerate https://github.com/nathanweeks/exonerate is one
      such tool
      5. Exonerate can align protein to genome or EST/Transcript to genome. NOT
      useful for all RNASeq results
      
      ---
      # Combining the evidence

      ![EviCombined](img/Yandell_Fig2.png)
      
      ---
      # Good evidence combiners
      1. EVM - Evidence Modeler - https://evidencemodeler.github.io/
      2. JIGSAW - https://ccb.jhu.edu/software/jigsaw/
      3. MAKER - http://www.yandell-lab.org/software/maker.html

      These tools combine the evidence data and predict the gene path
      that maximizes some scoring function. 
      
      ---
      #Some of best ''ab initio'' predictors
      ![EGASP](img/EGASP_Fig6a.png)
      1. Augustus - http://bioinf.uni-greifswald.de/augustus/
      2. Genemark.HMM - http://exon.gatech.edu/GeneMark/
      3. SNAP - http://korflab.ucdavis.edu/software.html

      EGASP: the human ENCODE Genome Annotation Assessment
      Project. Genome Biol. 2006;7 Suppl 1:S2.1-31
      ---
      # EGASP - breakdown of predictors
      
      ![EGASP](img/EGASP_Fig6b.png)

      ---
      # Putting the pieces together

      ![MAKER](img/MAKER_Apollo_view.jpg)

      ---
      # MAKER scoring gene models

      Maker scores a gene model based on the evidence which supports
      it.  The score is called AED or Annotation Edit Distance. The
      lower this number, the closer the evidence is to the final gene
      model

      Tutorials -
      Campbell et al. Genome annotation with MAKER. -
      http://dx.doi.org/10.1002/0471250953.bi0411s48 http://www.yandell-lab.org/publications/pdf/maker_current_protocols.pdf
      Yandel and Ence Beginner's guide to Genome annotation
      - http://dx.doi.org/10.1038/nrg3174
            

      ---
      # Aligning short sequencing reads

      Reads are aligned to the genome with tools such as Bowtie,
      Tophat, BWA.

      Tophat is useful for spliced RNAseq data as it can handle
      introns properly.  

      Some of these approaches will be obselete when long sequencing
      reads are the norm (e.g. PacBio, Nanopore)

      ---
      # Steps for running Tophat

      * Construct a genomic database and index it
      * Usually a FASTA file format
      * Run
      ```
      bowtie-index FASTAFILE GENOMENAME
      ```
      or
      ```
      bowtie2-index FASTAFILE GENOMENAME
      ```
      * Run tophat (there are many options)
      ```
      tophat GENOME file-withreads.fastq/fasta
      ```
      * Generates a BAM file (alignment) - can view this file with
      ```
      samtools view tophat_out/accepted_hits.bam | more
      1:4:4288:116769:Y	16	NcraOR74A_LGV:4113800..4140899	454
      50	50M *	0	0	TTTCTTAGCTTCAAAGTTCCTCCGAAC
      CCCAGGCTCTGTCGAGCTCGTTT	HHHHFHEHHFHHHHHHHHHHHHHHHFHHHHHHHHH
      HHHHHHHHHHHHHH      
      ```
      ---
      # Aligning proteins

      * BLASTX of proteins to the genome
      ```
      blastx -query file -db db -outfmt 6 -out results.BLASTX.tab
      ```
      * Results look like
      ```
      # TBLASTN 2.2.31+
      # Query: MGG_06016 | organism=Magnaporthe_oryzae_70-15 | product=conserved hypothetical protein | location=Mory70-15_SC24:706815-710252(+) | length=1080 | sequence_SO=supercontig | SO=protein_coding
      # Database: locus.fa
      # Fields: query id, subject id, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score
      # 2 hits found
      MGG_06016       NcraOR74A_LGV_4113800_4140899   52.01   596     267     6       224     809     6014    7774    2e-170  551
      MGG_06016       NcraOR74A_LGV_4113800_4140899   49.09   275     129     5       809     1079    7958    8761    4e-54   197
      ```
      ---
      # Aligning with splice aware information
      
      * Exonerate is tool that can align proteins but knows where

      ```
      exonerate -m p2g -q MGG_06016 -t locus.fa 
      Command line: [exonerate -m p2g -q MGG_06016 -t locus.fa]
      Hostname: [ZygoAir]

      C4 Alignment:
      ------------
         Query: MGG_06016 | organism=Magnaporthe_oryzae_70-15 | product=conserved hypothetical protein | location=Mory70-15_SC24:706815-710252(+) | length=1080 | sequence_SO=supercontig | SO=protein_coding
        Target: NcraOR74A_LGV_4113800_4140899
         Model: protein2genome:local
      Raw score: 1034
      Query range: 591 -> 999
      Target range: 7126 -> 8536

      592 :  ArgMetSerGlnSerLeuSerLeuSerArgSerArgMetLeuValProAlaAlaProArgIl :  612
             :::.........   ||||||::::::||||||:::   |||...      |||   :::
             LysPheGlyAsnProLeuSerMetAlaArgSerGlnAlaLeuThr---LeuAlaSerGlnSe
      7127 : aagtttgggaacccgctctccatggcccgctcacaagcccttaca---ttggcctcacagtc : 7184      
      ```

      ---
      #Exonerate generating GFF

      ```
      exonerate -m p2g -q MGG_06016 -t locus.fa  --showtargetgff
      Nc   exonerate  gene    7127    8536    1034    +       .       gene_id 1 ; sequence MGG_06016 ; gene_orientation +
      Nc   exonerate  cds     7127    7774    .       +       .
      Nc   exonerate  exon    7127    7774    .       +       .       insertions 0 ; deletions 2
      Nc   exonerate  splice5 7775    7776    .       +       .       intron_id 1 ; splice_site "gt"
      Nc   exonerate  intron  7775    7960    .       +       .       intron_id 1
      Nc   exonerate  splice3 7959    7960    .       +       .       intron_id 0 ; splice_site "ag"
      Nc   exonerate  cds     7961    8536    .       +       .
      Nc   exonerate  exon    7961    8536    .       +       .       insertions 9 ; deletions 1
      ```

      Without long lines so you can see extra info
      ```
      Nc   exonerate  gene    7127    8536    1034   gene_orientation +
      Nc   exonerate  cds     7127    7774    
      Nc   exonerate  exon    7127    7774    insertions 0 ; deletions 2
      Nc   exonerate  splice5 7775    7776    intron_id 1 ; splice_site "gt"
      Nc   exonerate  intron  7775    7960    intron_id 1
      Nc   exonerate  splice3 7959    7960    intron_id 0 ; splice_site "ag"
      Nc   exonerate  cds     7961    8536    
      Nc   exonerate  exon    7961    8536    insertions 9 ; deletions 1
      ```

      ---
      #Other types of genes

      * Non-protein coding genes are also in genomes      
      * tRNA, snoRNA using structural information http://lowelab.ucsc.edu/tRNAscan-SE/      
      * miRNA from sequence identity usually

      ![tRNA](img/tRNA_scan.png)
      
      ---
      # Genome annotation summmary

      * Overall automated tools to identify coding gene regions in
      genomes
      * Using prediction software and evidence based models

    </textarea>
    <script src="js/remark.js" type="text/javascript">
    </script>
    <script type="text/javascript">
      var slideshow = remark.create();
    </script>
  </body>
  </html>
